tensorboard/webapp/plugins/npmi/views/data_selection/metric_arithmetic/metric_arithmetic_element/metric_arithmetic_element_test.ts

```diff
@@ -0,0 +1,242 @@
+
+        expect(dispatchedActions).toEqual([
+          npmiActions.npmiChangeMetricFilter({
+            metric: 'npmi@test',
+            max: 1.0,
+            min: 0.2,
+            includeNaN: false,
+          }),
+        ]);
+      });
+
+      it('changes the filter to NaN', () => {
+        const fixture = TestBed.createComponent(
+          MetricArithmeticElementContainer
+        );
+        fixture.componentInstance.metric = 'npmi@test';
+        fixture.detectChanges();
+
+        const inputs = fixture.debugElement.queryAll(By.css(css.INPUT));
+        const input = inputs[0];
+        input.nativeElement.focus();
+        fixture.detectChanges();
+
+        input.nativeElement.value = 'NaN';
+        input.nativeElement.dispatchEvent(
+          new InputEvent('input', {data: 'NaN'})
+        );
+        fixture.detectChanges();
+
+        expect(dispatchedActions).toEqual([
+          npmiActions.npmiChangeMetricFilter({
+            metric: 'npmi@test',
+            max: 1.0,
+            min: -1.0,
+            includeNaN: true,
+          }),
+        ]);
+      });
+
+      it('does not change the filter when an incorrect value is entered', () => {
+        const fixture = TestBed.createComponent(
+          MetricArithmeticElementContainer
+        );
+        fixture.componentInstance.metric = 'npmi@test';
+        fixture.detectChanges();
+
+        const inputs = fixture.debugElement.queryAll(By.css(css.INPUT));
+        const input = inputs[0];
+        input.nativeElement.focus();
+        fixture.detectChanges();
+
+        input.nativeElement.value = '-';
+        input.nativeElement.dispatchEvent(new InputEvent('input', {data: '-'}));
+        fixture.detectChanges();
+
+        expect(dispatchedActions).toEqual([]);
+
+        const invalidInput = fixture.debugElement.query(
+          By.css(css.VALUE_INVALID)
+        );
+        expect(invalidInput).toBeTruthy();
+      });
+    });
+
+    describe('max input', () => {
+      it('changes the filter when interacting with the input', () => {
+        const fixture = TestBed.createComponent(
+          MetricArithmeticElementContainer
+        );
+        fixture.componentInstance.metric = 'npmi@test';
+        fixture.detectChanges();
+
+        const inputs = fixture.debugElement.queryAll(By.css(css.INPUT));
+        const input = inputs[1];
+        input.nativeElement.focus();
+        fixture.detectChanges();
+
+        input.nativeElement.value = '0.2';
+        input.nativeElement.dispatchEvent(
+          new InputEvent('input', {data: '0.2'})
+        );
+        fixture.detectChanges();
+
+        expect(dispatchedActions).toEqual([
+          npmiActions.npmiChangeMetricFilter({
+            metric: 'npmi@test',
+            max: 0.2,
+            min: -1.0,
+            includeNaN: false,
+          }),
+        ]);
+      });
+
+      it('does not change the filter when a value is entered that does not go with the min value', () => {
```

Comment:
i think there should be other specs that tests when the max value < min value and another case when both min and max are NaNs.